src/resmom/catch_child.c

```diff
@@ -1253,6 +1253,15 @@ send_obit(job *pjob, int exval)
 	struct resc_used_update rud;
 	pbs_list_head vnl_changes;
 
+	if (pjob->ji_execjob_end_hook_event_started)
```

Comment:
I think the code is becoming execjob_end hook specific. We would eventually make all the job lifecyle event hooks non-blocking for the mom, and thus the code is best if it stays generic. Please try to avoid hook event specific variables, and code structures. How about using something like "some job lifecyle events are being processed" flag, instead of a specific one. The other way to think about this to think of what "other" changes you would want to do if you had to make the execjob_launch hook non-blocking for example.

Reply:
Thank you Subhasis for the reviewing the code, I will make the changes accordingly.